sklearn/utils/tests/test_pprint.py

```diff
@@ -574,3 +574,36 @@ def set_params(self, **params):
     with config_context(print_changed_only=False):
         expected = "WithKWargs(a='something', b='unchanged', c='abcd', d=None)"
         assert expected == est.__repr__()
+
+def test_complexity_print_changed_only():
+    # Make sure `__repr__` is called the same amount of times
+    # whether `print_changed_only` is True or False
+    # Non-regression test for
+    # https://github.com/scikit-learn/scikit-learn/issues/18490
+
+    class ReprCounter(BaseEstimator):
+        nb_times_repr_called = 0
+        def __repr__(self):
+            ReprCounter.nb_times_repr_called += 1
+            return super().__repr__()
```

Comment:
Instead of defining this base class, would it be equivalent to just have `nb_times_repr_called` as a global variable and `DummyEstimator.__repr__` to be the current `__repr__` of `ReprCounter`?

Reply:
No, because the counter also needs to be increased when `__repr__` of `Pipeline` is called.